### PR TITLE
Provide examples for "how to" add components with varying complexity

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs!

I was inspired by this comment.

> It would be great to have a detailed explanation of how the components, namespaces and filenames work and interact with a specifiс complex example. Official docs ("Define the package information") are vague and too general. A CCI wiki page could be a good idea.

_Originally posted in https://github.com/conan-io/conan-center-index/pull/5062#discussion_r602884669_

Any suggestions are welcomed!

### CODE REVIEW

File: docs/how_to_add_packages.md [hunk 1]
```diff
@@ -162,11 +162,15 @@ If you are looking for header-only projects, you can take a look on [rapidjson](
 
 For C/C++ projects which use CMake for building, you can take a look on [szip](https://github.com/conan-io/conan-center-index/blob/master/recipes/szip/all/conanfile.py) and [recastnavigation](https://github.com/conan-io/conan-center-index/blob/master/recipes/recastnavigation/all/conanfile.py).
 
+Another common use case for CMake based projects, both header only and compiled, is _modeling components_ to match the `find_package` and export the correct targets from Conan's generators. A basic examples of this is [cpu_features](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpu_features/all/conanfile.py), a moderate/intermidiate example is [cpprestsdk](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpprestsdk/all/conanfile.py), and a very complex example is [OpenCV](https://github.com/conan-io/conan-center-index/blob/master/recipes/opencv/4.x/conanfile.py).
```
Comment (uilianries) [inline docs/how_to_add_packages.md:165]: Nice, 3 levels!

**Review by uilianries — APPROVED** [review]

File: docs/how_to_add_packages.md (hunk 1 above)
Comment (theirix) [inline docs/how_to_add_packages.md:165]: great list! just a small mistype - intermidiate -> intermediate

Comment (prince-chrismc) [reply docs/how_to_add_packages.md:165]:
```suggestion
Another common use case for CMake based projects, both header only and compiled, is _modeling components_ to match the `find_package` and export the correct targets from Conan's generators. A basic examples of this is [cpu_features](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpu_features/all/conanfile.py), a moderate/intermediate example is [cpprestsdk](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpprestsdk/all/conanfile.py), and a very complex example is [OpenCV](https://github.com/conan-io/conan-center-index/blob/master/recipes/opencv/4.x/conanfile.py).
```

**Review by theirix — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by rockdreamer — DISMISSED** [review]

**Comment by jgsogo** [comment]
We have a conflict here, @prince-chrismc

**Comment by prince-chrismc** [comment]
Updated, please review 💟

**Review by rockdreamer — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
Thank you!

**Review by SSE4 — APPROVED** [review]

**Pull request merged**